app/models/user.rb

```diff
@@ -3,19 +3,22 @@ class User < ApplicationRecord
   has_many :clones, dependent: :destroy
 
   def self.from_omniauth(auth)
-    User.find_or_create_by(uid: auth.uid) do |user|
-      is_admin = false
+    user = User.find_by(uid: auth.uid)
+    is_admin = false
+    if user.nil?
+      user = User.create(uid: auth.uid)
       client = Octokit::Client.new(access_token: auth.credentials.token)
       approved_orgs = ApprovedOrganizations.all
       found_org = client.orgs.find { |org| org.login.in?(approved_orgs) }
 
       is_admin = true if found_org
-
       user.admin = is_admin
-      user.nickname = auth.info.nickname
-      user.token = auth.credentials.token
-      user.name = auth.info.name
-      user.email = auth.info.email
     end
+    user.nickname = auth.info.nickname
+    user.token = auth.credentials.token
+    user.name = auth.info.name
+    user.email = auth.info.email
+    user.save
+    user
```

Comment:
moving these pieces out of the find_or_create_by allows the email/token/name/etc to be updated every time the user logs in; this fixed the problem for two students yesterday who had stale tokens.